src/domain/interfaces/prices-api.interface.ts

```diff
@@ -15,4 +15,10 @@ export interface IPricesApi {
   }): Promise<AssetPrice>;
 
   getFiatCodes(): Promise<string[]>;
+
+  registerNotFoundTokenPrice(args: {
+    chainName: string;
+    tokenAddress: string;
+    fiatCode: string;
+  }): Promise<void>;
```

Comment:
Is it possible to move this logic so that it's internal to the datasource? I don't think the domain should be the one deciding when to call this.

Reply:
Yes, I also thought about that. Finally I left this on the domain because as I commented in another comment, there's no way to check if the response actually contains the price without processing it (the status code is always 200). So moving this to the datasources layer would imply moving the response validation there.

Reply:
Understood! Can you leave a comment on this function explaining why it exists? Compared to the other functions this seems a bit a out of place 😬

Reply:
Added in [7538cd9](https://github.com/safe-global/safe-client-gateway/pull/828/commits/7538cd923d47b7eaefbfde53d1482a36af131894)

I hope it's understandable 🙂 Thanks!